src/tasks/installers/olm.ts

```diff
@@ -0,0 +1,327 @@
+/*********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import Command from '@oclif/command'
+import Listr = require('listr')
+
+import { KubeHelper } from '../../api/kube'
+import { CatalogSource, Subscription } from '../../api/typings/olm'
+import { CheOLMChannel, DEFAULT_CHE_IMAGE, defaultKubernetesMarketPlaceNamespace, defaultOLMKubernetesNamespace, defaultOpenshiftMarketPlaceNamespace, kubernetesApplicationPreviewRegistryNamespace, openshiftApplicationPreviewRegistryNamespace } from '../../constants'
+import { isKubernetesPlatformFamily } from '../../util'
+
+import { checkPreCreatedTls, checkTlsSertificate, copyOperatorResources, createEclipeCheCluster, createNamespaceTask } from './common-tasks'
+
+export class OLMTasks {
+  private readonly operatorSourceName = 'eclipse-che'
+  private readonly subscriptionName = 'eclipse-che-subscription'
+  private readonly operatorGroupName = 'che-operator-group'
+  private readonly packageNamePrefix = 'eclipse-che-preview-'
+  private readonly channel = this.getDefaultChannel()
+
+  /**
+   * Returns list of tasks which perform preflight platform checks.
+   */
+  startTasks(flags: any, command: Command): Listr {
+    const kube = new KubeHelper(flags)
+    return new Listr([
+      this.isOlmPreInstalledTask(command, kube),
+      copyOperatorResources(flags, command.config.cacheDir),
+      createNamespaceTask(flags),
+      checkPreCreatedTls(flags, kube),
+      checkTlsSertificate(flags),
+      {
+        title: 'Create operator group',
+        task: async (task: any) => {
+          if (await kube.operatorGroupExists(this.operatorGroupName, flags.chenamespace)) {
+            task.title = `${task.title}...It already exists.`
+          } else {
+            await kube.createOperatorGroup(this.operatorGroupName, flags.chenamespace)
+            task.title = `${task.title}...OK`
```

Comment:
I think we usually use `done`. Please check chectl output.

Reply:
Looks like this... But for delete tasks we are using `...OK` for some unknown for me reasons..